ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/AtomicActionRecoveryModule.java

```diff
@@ -223,17 +231,39 @@ private void processTransactionsStatus()
                 try {
                     if (_recoveryStore.currentState(currentUid,
                             _transactionType) != StateStatus.OS_UNKNOWN) {
-                        doRecoverTransaction(currentUid);
+                        RecoverAtomicAction rcvAtomicAction = doRecoverTransaction(currentUid);
+
+                        /*
+                         * If the current AtomicAction has been recovered, its StateStatus should be OS_UNKNOWN.
+                         * If that is not the case, it means that the current AtomicAction still needs to be recovered
+                         */
+                        if (_recoveryStore.currentState(currentUid, _transactionType) != StateStatus.OS_UNKNOWN) {
+                            if (rcvAtomicAction.hasHeuristicParticipants() &&
+                                    !recoveryPropertyManager.getRecoveryEnvironmentBean().isWaitForHeuristicsDuringRecovery()) {
+                                tsLogger.i18NLogger.heuristic_atomic_action_silenced(currentUid);
```

Comment:
Perhaps this could be trace instead? I think that would be OK wouldn't it given it seems it's the users intention to ignore them but still providing a way to review what went on during a suspend (if tracing is enabled).

Reply:
Or log it just once which is enough for admins to figure out which resource they need to investigate.

Reply:
This branch of the code is for when there are heuristics but the user has asked not to waited for them so it should only be logged once (assuming there are no other recovery modules with work to do).

Reply:
This has been discussed and the heuristic switch has been removed. Shall we resolve this conversation as well?

Reply:
Sorry @mmusgrov, can we resolve this conversation or is there anything I missed?